pkg/metric/writer_cw.go

```diff
@@ -98,11 +116,32 @@ func (w *cloudwatchWriter) WriteOne(ctx context.Context, data *Datum) {
 	w.Write(ctx, Data{data})
 }
 
-func (w *cloudwatchWriter) Write(ctx context.Context, batch Data) {
+func (w *cloudwatchWriter) Write(applicationCtx context.Context, batch Data) {
 	if len(batch) == 0 {
 		return
 	}
 
+	timeout := w.writeTimeout
+	if timeout <= 0 {
+		timeout = defaultWriteTimeout
+	}
+
+	manualCtx, cancel := exec.WithDelayedCancelContext(applicationCtx, timeout)
+
+	cfn, manualCtx := coffin.WithContext(manualCtx)
```

Comment:
What is the use of the coffin here? It will never fail before the write method returns or the delayed context is cancelled, so why not use that directly?